docs/proposals/simulation/simulation.md

```diff
@@ -0,0 +1,165 @@
+
+It will provide Ianvs with docker in docker system emulation features.
+
+## Design Details
+
+### Dataset automatic download function
+
+The current `testenv.yaml` file is as follows
+
+```yaml
+testenv:
+  # dataset configuration
+  dataset:
+    # the url address of train dataset index; string type;
+    train_url: "/ianvs/dataset/train_data/index.txt"
+    # the url address of test dataset index; string type;
+    test_url: "/ianvs/dataset/test_data/index.txt"
+
+  # metrics configuration for test case's evaluation; list type;
+  metrics:
+      # metric name; string type;
+    - name: "f1_score"
+      # the url address of python file
+      url: "./examples/pcb-aoi/singletask_learning_bench/testenv/f1_score.py"
+```
+
+For the convenience of developers, the expected testenv.yaml file is as follows:
+
+- The user fills in the `dataset_url`, e.g. HUAWEI CLOUD OBS link.
+- The user fills in `dataset_path`, where the dataset wants to be saved.
+
+```yaml
+testenv:
+  # dataset configuration
+  dataset:
+    # the url address of dataset with zip file; string type;
+    dataset_url: "https://kubeedge.obs.cn-north-1.myhuaweicloud.com:443/ianvs/pcb-aoi/dataset.zip"
+    # the url address of saving dataset
+    dataset_path: "./dataset/"
+
+  # metrics configuration for test case's evaluation; list type;
+  metrics:
+      # metric name; string type;
+    - name: "f1_score"
+      # the url address of python file
+      url: "./examples/pcb-aoi/singletask_learning_bench/testenv/f1_score.py"
+```
+
+The expected implementation method is to modify `dataset.py` in the `testenvmanager` module as follows:
+
+1. Add `self.dataset_url`, `self.dataset_path`
+2. Add `download_save()` function
+
+![](https://github.com/iszhyang/images/blob/main/ianvs/dataset-download.png?raw=true)
+
+### Semantic segmentation dataset and corresponding algorithm
+
+dataset: [cityscapes](https://www.cityscapes-dataset.com)
+Baseline algorithm: [RFNet](https://github.com/AHupuJR/RFNet)
+
+We will provide the cityscapes dataset and corresponding methods for ianvs. The specific operation is as follows:
+
+- dataset processing
+  - generate the index file for train and test
+- dataset download link
+  - provide the download link of cityscapes dataset in HUAWEI Cloud OBS.
+- Baseline algorithm
+  ![](https://github.com/iszhyang/images/blob/main/ianvs/rfnet.png?raw=true)
+
+### Industrial Distributed Collaborative System Simulation
+
+Provide ianvs with the feature of industrial distributed system simulation.
+#### Framework
+![img](images/simulation_framework.jpg)
+
+In the framwork of ianvs, the `simulation controller` is the core module of system simulation. The `simulation controller` has been supplemented, which build and deploy local edge-cloud simulation environment with K8s.
+
+![img](images/simulation_controller.jpg)
+
+The models in `simulation controller` are as follows:
+- The `Simulation System Administrator` is used to
+  1. parse the system config(simulation)
```

Comment:
Before creating a simulation environment, you need to check whether the environment has the startup conditions, such as whether tools such as docker and kind are installed, and whether the CPU and memory are sufficient.

Reply:
Your text description needs to be consistent with the description in the flowchart.

Reply:
Thanks. I have modified it according to your suggestion.